research-software-directory/RSD-as-a-service
# 393 meta pages

# Setup meta pages

Closes #393 

Changes proposed in this pull request:
* Admins are able to crate custom markdown pages
* They have additional option to manage these pages
* No default pages are created. 
* The position of the page in the list can be changed using drag-and-drop support

**NOTE 1: there is a bug in DND library we use. After completely reloading the page, drag-and-drop functionality is not present. Further investigation is needed**.

**Note 2: after enabling specific page you will need to save and reload RSD in the browser to be able to see new link in the footer**.

How to test:
* `docker-compose down --volumes` to remove old
* `docker-compose build` to build
*  define `professor3` as rsd admin in env file
```env
RSD_ADMIN_EMAIL_LIST=isaacnewton@university-example.org
```
* `docker-compose up` to start solution
* visit http://localhost and confirm no meta pages are listed in the footer
![image](https://user-images.githubusercontent.com/9204081/179418971-380ce52f-8fe4-4a0b-83b4-7bef7b10de5b.png)
* login as `professor3`, if you define rsd admin as suggested. Confirm the additional option in the menu "Administration"
![image](https://user-images.githubusercontent.com/9204081/179419039-e263d318-0543-4d29-bb09-5d327bf300f3.png)
* click on the option and you will be redirected to new page (see image)
![image](https://user-images.githubusercontent.com/9204081/179782258-8dbfc46a-08f5-4cfb-966e-e9b3ed15328e.png)
* Add, edit and delete pages. 
![image](https://user-images.githubusercontent.com/9204081/179781827-238eeed0-96fa-4138-b580-e9dce3a4ed81.png)

PR Checklist:
